### DISCUSSION TITLE
vmware images

### DISCUSSION THREAD

**Discussion by johdcmlaker2000** [tetrane/reven-community#17] [Q&A] [answered]
Is vmware images supported? I'm addding .vmdk image previously full cloned vm and getting "Last error: Not a valid Qemu VM file" error.

**Comment by johdcmlaker2000** [comment]
my case
https://bugs.launchpad.net/qemu/+bug/1793016

**Comment by johdcmlaker2000** [comment]
test@debian:~/VMs/Clone of Windows 7 x64 new$ qemu-img convert -f vmdk -O qcow2 Windows\ 7\ x64\ new-cl1-flat.vmdk image.qcow
qemu-img: Could not open 'Windows 7 x64 new-cl1-flat.vmdk': invalid VMDK image descriptor

test@debian:~/VMs/Clone of Windows 7 x64 new$ qemu-img convert -O qcow2 Windows\ 7\ x64\ new-cl1-flat.vmdk image.qcow

**Comment by johdcmlaker2000** [comment]
no way to upload raw images from project manager?

**Comment by Quentin01** [comment, marked as answer]
Hi,

Reading the error message, it looks like the importation failed because the 'raw' format is not enabled for import in your REVEN installation. See [here](https://doc.tetrane.com/free/latest/Managing-VMs/Adding-VMs/Misc.html?highlight=QUASAR_QEMU#supported-disk-formats) for the list of default-supported formats.

You can actually add the raw format in `QUASAR_QEMU_SCAN_FORMATS` in the `~/.config/tetrane/quasar.py` (see [here](https://doc.tetrane.com/free/latest/Installation/Server/Native.html#configuring-the-package) for how to do this).

1. Edit `quasar.py` to add:
```python
# quasar.py
QUASAR_QEMU_SCAN_FORMATS = [
    "qcow",
    "qcow2",
    "vdi",
    "vmdk",
    "vhdx",
    "raw",
]
```
2. Then restart the Project Manager.
3. The importation of the raw file should now succeed.

Note that adding the `raw` format in the scan formats will result in possibly many files being listed in the "Register VM" page, because QEMU considers any unknown format as "raw".

I guess we could make that error message clearer in the future :sweat_smile:.
